dace/libraries/blas/utility/reductions.py

```diff
@@ -0,0 +1,845 @@
+        dst_conn='prevCon',
+        memlet=Memlet.simple(
+            partial_in.data, '0' if partial_width == 1 else 'i % {0}'.format(partial_width),
+        )
+    )
+
+    store_state.add_memlet_path(
+        store_task, partial_res,
+        src_conn='outCon',
+        memlet=Memlet.simple(
+            partial_res.data, '0' if partial_width == 1 else 'i % {0}'.format(partial_width),
+            # wcr_str=reductionLambda,
+        )
+    )
+
+
+    inner_sdfg.fill_scope_connectors()
+    inner_sdfg.add_edge(init_state, compute_state, InterstateEdge(None))
+    inner_sdfg.add_edge(compute_state, store_state, InterstateEdge(None))
+
+
+    # CONNECT NESTED
+    # -----------------------
+
+
+    nested_sdfg = state.add_nested_sdfg(
+        inner_sdfg,
+        sdfg,
+        {'inner_x', 'inner_y'},
+        {'partial_out'}
+
+    )
+
+    state.add_memlet_path(
+        x_in, outerMap_entry, nested_sdfg,
+        dst_conn='inner_x',
+        memlet=Memlet.simple(
+            x_in.data, '0:{}'.format(srcSize/veclen)
+        )
+    )
+
+    state.add_memlet_path(
+        y_in, outerMap_entry, nested_sdfg,
+        dst_conn='inner_y',
+        memlet=Memlet.simple(
+            y_in.data, '0:{}'.format(srcSize/veclen)
+        )
+    )
+
+    state.add_memlet_path(
+        nested_sdfg, outerMap_exit, buf_in,
+        src_conn='partial_out',
+        memlet=Memlet.simple(
+            buf_in.data, '0:{}'.format((max(partial_width, 2))),
+            # wcr_str=reductionLambda
+        )
+    )
+
+
+
+
+
+
+
+def fpga_linear_result_reduction(
+        state,
+        src,
+        dest,
+        dtype,
+        srcSize,
+        reductionLambda='lambda a, b: a + b',
+        toMem=False
+    ):
+    """
+    Reduces an input array linearly into a scalar
+    with the given reduction function
+    """
+
+    red_buf = state.add_read(src)
+    buf_out = 0
+    if toMem:
+        buf_out = state.add_write(dest)
+    else:
+        buf_out = state.add_stream(
+            dest,
+            dtype,
+            buffer_size=32,
+            storage=dtypes.StorageType.FPGA_Local
+        )
+
+    outerMap_entry, outerMap_exit = state.add_map(
+        'reduce_map',
+        dict(i="0:{0}".format(srcSize)),
+        schedule=dtypes.ScheduleType.FPGA_Device,
+        unroll=True
+    )
+
+    innerTasklet = state.add_tasklet(
+        'reduce_task',
+        ['inCon'],
+        ['outCon'],
+        'outCon = inCon'
+    )
+
+    state.add_memlet_path(
+        red_buf, outerMap_entry, innerTasklet,
+        dst_conn='inCon',
+        memlet=Memlet.simple(
+            red_buf.data, 'i'
+        )
+    )
+
+    state.add_memlet_path(
+        innerTasklet, outerMap_exit, buf_out,
+        src_conn='outCon',
+        memlet=Memlet.simple(
+            buf_out.data, '0',
+            wcr_str=reductionLambda,
+        )
+    )
+
+
+
+
+
+def fpga_make_matrixPartialReduction(dtype, nTile, mTile, partialWidth, n, m, veclen, a, b):
```

Comment:
Please do a pass to use our naming convention for `variable_names` and `function_names` (i.e., search for `[a-z][A-Z]`).

Reply:
Fixed